### PR TITLE
blast improvements

### PR DESCRIPTION
**Pull request by bernt-matthias**
-  properly quote cached databases 
- add wrapper for get_species_taxids.xml (will only work once https://github.com/bioconda/bioconda-recipes/pull/25680 is merged)
- allow the input of taxid lists

may supersede https://github.com/peterjc/galaxy_blast/pull/126

### CODE REVIEW

**Comment by bernt-matthias** [comment]
ping @peterjc

**Comment by peterjc** [comment]
I don't think I'd noticed ``get_species_taxids.sh`` before, but bioconda installs it.

**Comment by peterjc** [comment]
https://www.ncbi.nlm.nih.gov/books/NBK279669/ is a nice find. I remember banging my head against the wall with that year ago, and concluding it was not possible

**Comment by peterjc** [comment]
This seems sensible, needs a tiny update in ``tools/ncbi_blast_plus/README.rst`` for change log for ``2.10.1+galaxy1`` (which I should probably release now), and your name if you would like.

**Review by peterjc — APPROVED** [review]

**Comment by peterjc** [comment]
Are you familiar with "git rebase"? Either that "Resolve conflicts" button will let you sort out the change log conflict:

```
2.10.1+galaxy1 - Make locally installed database selector non-optional.
```

**Comment by peterjc** [comment]
Did you check the option to allow maintainers to edit this branch? I wanted to tweak the README...

https://github.com/peterjc/galaxy_blast/pull/new/blast_imp

**Comment by bernt-matthias** [comment]
> Did you check the option to allow maintainers to edit this branch?

Yes

**Comment by peterjc** [comment]
Hmm. I must have done something wrong with adding the remote or my password then...

**Comment by bernt-matthias** [comment]
> Hmm. I must have done something wrong with adding the remote or my password then...

Then PR to my branch? Or use suggestions in the "files changed" tab?

I'm currently debugging the validators ..

**Comment by peterjc** [comment]
Got it, remote URL ``git@github.com:bernt-matthias/galaxy_blast.git`` works which is not one of GitHub's suggestions.

**Comment by peterjc** [comment]
I need to setup pre-commit on this repository, failed my own check: "Tabs are bad, please use four spaces in RST files."

File: tools/ncbi_blast_plus/ncbi_macros.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <macros>
     <token name="@TOOL_VERSION@">2.10.1</token>
-    <token name="@VERSION_SUFFIX@">1</token>
+    <token name="@VERSION_SUFFIX@">2</token>
```
Comment (peterjc) [inline tools/ncbi_blast_plus/ncbi_macros.xml:3]: We don't need a bump, suffix 1 has not yet been uploaded to the main tool shed

**Comment by bernt-matthias** [comment]
@peterjc  Should be good to go?

**Comment by peterjc** [comment]
Can you double check the ``README.rst``, looks like a duplication in the change log via the merging?

File: tools/ncbi_blast_plus/README.rst [hunk 2]
```diff
@@ -136,7 +136,22 @@ a galaxy specific suffix which gets reset to zero with each new BLAST version:
...
+                 by species taxids given in a file.
+               - Properly quote cached databases
 2.10.1+galaxy1 - Make locally installed database selector non-optional.
+               - Add tool "NCBI get species taxids" wrapping NCBI's
+                 ``get_species_taxids.sh`` script which determines all species
+                 taxids below a certain taxon, see
+                 https://www.ncbi.nlm.nih.gov/books/NBK546209/
+               - Can restrict BLAST searches taxonomically (using the
+                  ``-taxidlist`` and ``-negative_taxidlist`` options).
+               - Properly quote multiple database filenames with spaces.
```
Comment (bernt-matthias) [inline tools/ncbi_blast_plus/README.rst:154]:
```suggestion
2.10.1+galaxy2 - Add tool `NCBI get species taxids` that wraps NCBI's
                 `get_species_taxids.sh` script
                 (https://www.ncbi.nlm.nih.gov/books/NBK546209/).
                 It allows to determine all species taxids below a certain
                 Taxon.
               - Add the possibility to restrict BLAST searches taxonomically
                 by species taxids given in a file.
               - Properly quote cached databases
2.10.1+galaxy1 - Make locally installed database selector non-optional.
```

**Comment by peterjc** [comment]
This would still be ``2.10.1+galaxy1`` not ``2.10.1+galaxy1`` (because I never got round to doing that release).

File: tools/ncbi_blast_plus/README.rst [hunk 3]
```diff
@@ -136,6 +136,14 @@ a galaxy specific suffix which gets reset to zero with each new BLAST version:
...
 -------------- ---------------------------------------------------------------
+2.10.1+galaxy2 - Add tool `NCBI get species taxids` that wraps NCBI's
+                 `get_species_taxids.sh` script
+                 (https://www.ncbi.nlm.nih.gov/books/NBK546209/).
+                 It allows to determine all species taxids below a certain
+                 Taxon.
+               - Add the possibility to restrict BLAST searches taxonomically
+                 by species taxids given in a file.
+               - Properly quote cached databases
 2.10.1+galaxy1 - Make locally installed database selector non-optional.
```
Comment (bernt-matthias) [inline tools/ncbi_blast_plus/README.rst:147]:
```suggestion
2.10.1+galaxy1 - Add tool `NCBI get species taxids` that wraps NCBI's
                 `get_species_taxids.sh` script
                 (https://www.ncbi.nlm.nih.gov/books/NBK546209/).
                 It allows to determine all species taxids below a certain
                 Taxon.
               - Add the possibility to restrict BLAST searches taxonomically
                 by species taxids given in a file.
               - Properly quote cached databases
               - Make locally installed database selector non-optional.
```

**Comment by peterjc** [comment]
Thank you!

GitHub's confirm squash and merge button does not seem to be working right now. There's an actions outage listed on https://www.githubstatus.com/ so I should try this again later I think...

**Review by peterjc — APPROVED** [review]

**Comment by peterjc** [comment]
Test and main Tool Shed Updated, thank you.

**Pull request merged**